tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+The settings of the *`target-type`* are processed first; then the settings of the *`build-type`* that potentially overwrite the *`target-type`* settings.
+
+YML structure:
+
+```yml
+target-types:          # Start a list of target type declarations
+  - type:              # name of the target type (required)
+    board:             # board specification (optional)
+    device:            # device specification (optional)         
+    processor:         # processor specific settings (optional)
+    compiler:          # toolchain specification (optional) 
+    optimize:          # optimize level for code generation (optional)
+    debug:             # generation of debug information (optional)
+    defines:           # define symbol settings for code generation (optional).
+    undefines:         # remove define symbol settings for code generation (optional).
+    add-paths:         # additional include file paths (optional).
+    del-paths:         # remove specific include file paths (optional). 
+    misc:              # Literal tool-specific controls
+
+build-types:           # Start a list of build type declarations
+  - type:              # name of the build type (required)
+    processor:         # processor specific settings (optional)
+    compiler:          # toolchain specification (optional) 
+    optimize:          # optimize level for code generation (optional)
+    debug:             # generation of debug information (optional)
+    defines:           # define symbol settings for code generation (optional).
+    undefines:         # remove define symbol settings for code generation (optional).
+    add-paths:         # additional include file paths (optional).
+    del-paths:         # remove specific include file paths (optional). 
+    misc:              # Literal tool-specific controls
+```
+
+**Example:**
+
+```yml
+target-types:
+  - type: Board
+    board: NUCLEO-L552ZE-Q       # board specifies indirectly also the device
+
+  - type: Production-HW
+    device: STM32L552RC          # specifies device
+      
+build-types:
+  - type: Debug
+    optimize: debug
+    debug: on
+
+  - type: Test
+    optimize: max
+    debug: on
+```
+
+The `board:`, `device:`, and `processor:` settings are used to configure the code translation for the toolchain.  These settings are processed in the following order:
+
+1. `board:` relates to a BSP software pack that defines board parameters, including the [mounted device](https://arm-software.github.io/CMSIS_5/Pack/html/pdsc_boards_pg.html#element_board_mountedDevice).  If `board:` is not specified, a `device:` most be specified.
+2. `device:` defines the target device.  If `board:` is specified, the `device:` setting can be used to overwrite the device or specify the processor core used.
+3. `processor:` overwrites default settings for code generation, such as endianess, TrustZone mode, or disable Floating Point code generation.
```

Comment:
unclear to me

A TrustZone mode is a virtualization, still it is one core.
I would prefer talking about HW execution unit when it comes to TZ mode.
And this is important at execution context level.

To me, a processor can have one or several cores and each core can have several HW execution units (virtualization).